### ISSUE TITLE
Messaging service encrypts GROUP0 and GET_SCHEMA_VERSION messages regardless of DC/RACK

### ISSUE THREAD

**Issue by xemul** [scylladb/scylladb#11483]
The code

```c++
    auto must_encrypt = [&] {
        if (_cfg.encrypt == encrypt_what::none) {
            return false;
        }
        if (_cfg.encrypt == encrypt_what::all) {
            return true;
        }

        // if we have dc/rack encryption but this is gossip, we should
        // use tls anyway, to avoid having mismatched ideas on which 
        // group we/client are in. 
        if (verb >= messaging_verb::GOSSIP_DIGEST_SYN && verb <= messaging_verb::GOSSIP_SHUTDOWN) {
            return true;
        }
        ...
```

However, since encryption is turned ON/OFF on per-index basis, all verbs that share index with gossiper verbs are also encrypted. The verbs are:

```c++
    switch (verb) {
    // GET_SCHEMA_VERSION is sent from read/mutate verbs so should be
    // sent on a different connection to avoid potential deadlocks
    // as well as reduce latency as there are potentially many requests
    // blocked on schema version request.
    case messaging_verb::GOSSIP_DIGEST_SYN:
    case messaging_verb::GOSSIP_DIGEST_ACK:
    case messaging_verb::GOSSIP_DIGEST_ACK2:
    case messaging_verb::GOSSIP_SHUTDOWN:
    case messaging_verb::GOSSIP_ECHO:
    case messaging_verb::GOSSIP_GET_ENDPOINT_STATES:
    case messaging_verb::GET_SCHEMA_VERSION:
    // Raft peer exchange is mainly running at boot, but still
    // should not be blocked by any data requests.
    case messaging_verb::GROUP0_PEER_EXCHANGE:
    case messaging_verb::GROUP0_MODIFY_CONFIG:
    case messaging_verb::GET_GROUP0_UPGRADE_STATE:
        return 0;
```

@gleb-cloudius , is this something to worry about?

**Comment by gleb-cloudius** [comment]
If we encrypt more than needed for verbs that are not on the data path it should not be a problem. But I see different problem here. If the first verb that is used is not one of gossiper's then the connection is opened without encryption and subsequent communication is not encrypted for the gossiper verbs as well. If this is a problem? I am not sure since I do not understand the comment in `must_encrypt`. @elcallio should know better.

**Comment by xemul** [comment]
The comment as far as I understand it is the chicken and egg problem. To get dc/rack info we need to gossip, to gossip we need to start sending RPCs, to start sending RPCs we need to know dc/rack. So this place cuts this loop by encrypting gossiper unconditionally.

The same issue exists for tcp-nodelay, it was solved the same way, but then was broken (#11465).
And for compression as well (#11514)
